### ISSUE TITLE
Is there any way to use termux:X11 after i sign with my own keys? 

### ISSUE THREAD

**Issue by Anonymous2716** [termux/termux-x11#328]
I get:
```
Signatures of this loader and target application com.termux.x11 do not match.
Please, reinstall both termux-x11 package and Termux:X11 application from the same source
```

What should be the workaround? Instead of rebuilding and including my own keys there ? 

The loader apk doesn't contain any signature. What is it comparing then?

I'd like to know these Info's.

**Comment by twaik** [comment]
Just sign both apk's with the same signature.

**Comment by Anonymous2716** [comment]
@twaik I tried that. And I used apksigner but it terminates immediately after execution. And I get exit code 137 (echo $?).

**Comment by twaik** [comment]
Actually you should get apk with the same signature if you are compiling any of apk with `./gradlew assembleDebug`

**Comment by Anonymous2716** [comment]
> Actually you should get apk with the same signature if you are compiling any of apk with `./gradlew assembleDebug`

Seems like this is what I need to do.

**Comment by twaik** [comment]
Original files are signed with [this key](https://github.com/termux/termux-x11/blob/master/app/testkey_untrusted.jks) with [these credentials](https://github.com/termux/termux-x11/blob/75a6529121d265217ef32b6a28cd6d5312ef9367/app/build.gradle#L27-L34).

**Comment by Anonymous2716** [comment]
That I can see. But I was wondering if I can use my own keys without compiling from source. Also the loader apk doesn't have any signatures (RSA,SF,MF)in the META-INF file. Then what signature is the app  comparing  with termux:x11 apk? . maybe i don't fully understand yet that's why I need clarification.

**Comment by twaik** [comment]
That is the whole signature checking code. [Link](https://github.com/termux/termux-x11/blob/75a6529121d265217ef32b6a28cd6d5312ef9367/shell-loader/src/main/java/com/termux/x11/Loader.java#L41-L50). So signing both apks with same signature should work.

**Comment by Anonymous2716** [comment]
> So signing both apks with same signature should work.

 My thts exactly. But here is what happens after signing. 

```
.../libexec/termux-x11 $ termux-x11 :1
Killed                                                           
.../libexec/termux-x11 $ echo $?
137                                                                    
.../libexec/termux-x11 $
```
 i will Start over later . and will see what happens.

**Comment by twaik** [comment]
Check logcat output.

**Comment by Anonymous2716** [comment]
Unfortunately I don't have adb access right now . and I am on Android 9. I will update the logcats few hours later when i get in touch with a PC.

**Comment by twaik** [comment]
Start logcat from termux commandline.

**Comment by Anonymous2716** [comment]
Unfortunately I don't have adb access right now . and I am on Android 9. I will update the logcats few hours later when i get in touch with a PC.

**Comment by Anonymous2716** [comment]
Termux logcats:
```
05-27 05:13:56.568 26436 26436 E AndroidRuntime: FATAL EXCEPTION: main
05-27 05:13:56.568 26436 26436 E AndroidRuntime: PID: 26436
05-27 05:13:56.568 26436 26436 E AndroidRuntime: java.lang.UnsatisfiedLinkError: dlopen failed: library "/data/app/com.termux.x11-tru9m7s7szKY8Ou2FH0AIA==/base.apk!/lib/arm64-v8a/libXlorie.so" not found
05-27 05:13:56.568 26436 26436 E AndroidRuntime:        at java.lang.Runtime.loadLibrary0(Runtime.java:1016)
05-27 05:13:56.568 26436 26436 E AndroidRuntime:        at java.lang.System.loadLibrary(System.java:1669)
05-27 05:13:56.568 26436 26436 E AndroidRuntime:        at com.termux.x11.CmdEntryPoint.<clinit>(CmdEntryPoint.java:95)
05-27 05:13:56.568 26436 26436 E AndroidRuntime:        at java.lang.Class.classForName(Native Method)
05-27 05:13:56.568 26436 26436 E AndroidRuntime:        at java.lang.Class.forName(Class.java:453)
05-27 05:13:56.568 26436 26436 E AndroidRuntime:        at com.termux.x11.Loader.main(Loader.java:59)
05-27 05:13:56.568 26436 26436 E AndroidRuntime:        at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
05-27 05:13:56.568 26436 26436 E AndroidRuntime:        at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:340)
05-27 05:13:56.569 26436 26436 E AndroidRuntime: Error reporting crash
05-27 05:13:56.569 26436 26436 E AndroidRuntime: java.lang.RuntimeException: Bad file descriptor
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at android.os.BinderProxy.transactNative(Native Method)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at android.os.BinderProxy.transact(Binder.java:1137)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at android.os.ServiceManagerProxy.getService(ServiceManagerNative.java:125)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at android.os.ServiceManager.rawGetService(ServiceManager.java:253)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at android.os.ServiceManager.getService(ServiceManager.java:124)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at android.app.ActivityManager$1.create(ActivityManager.java:4133)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at android.app.ActivityManager$1.create(ActivityManager.java:4130)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at android.util.Singleton.get(Singleton.java:34)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at android.app.ActivityManager.getService(ActivityManager.java:4126)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$KillApplicationHandler.uncaughtException(RuntimeInit.java:143)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1068)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1063)
05-27 05:13:56.569 26436 26436 E AndroidRuntime:        at java.lang.Thread.dispatchUncaughtException(Thread.java:1955)
05-27 05:16:11.868 26709 26709 F libc    : Fatal signal 31 (SIGSYS), code 1 (SYS_SECCOMP) in tid 26709 (main), pid 26709 (main)
05-27 05:16:11.930 26712 26712 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
05-27 05:16:11.930 26712 26712 F DEBUG   : Build fingerprint: 'INTENTIONALLY REMOVED'-
05-27 05:16:11.930 26712 26712 F DEBUG   : Revision: '0'
05-27 05:16:11.930 26712 26712 F DEBUG   : ABI: 'arm64'
05-27 05:16:11.931 26712 26712 F DEBUG   : pid: 26709, tid: 26709, name: main  >>> /system/bin/app_process <<<
05-27 05:16:11.931 26712 26712 F DEBUG   : signal 31 (SIGSYS), code 1 (SYS_SECCOMP), fault addr --------
05-27 05:16:11.931 26712 26712 F DEBUG   : Cause: seccomp prevented call to disallowed arm64 system call 144
05-27 05:16:11.931 26712 26712 F DEBUG   :     x0  00000000000003e8  x1  0000007ff6196e70  x2  0000000000000000  x3  0000000000000000
05-27 05:16:11.931 26712 26712 F DEBUG   :     x4  0000007373a6c558  x5  0000007373a6c558  x6  0000007373a6c558  x7  0000000003f65782
05-27 05:16:11.931 26712 26712 F DEBUG   :     x8  0000000000000090  x9  0000000000000001  x10 0000000000004001  x11 0000000000000000
05-27 05:16:11.931 26712 26712 F DEBUG   :     x12 0000007373a6c558  x13 0000000000000000  x14 00000000ffffffff  x15 0000007ff6196d08
05-27 05:16:11.931 26712 26712 F DEBUG   :     x16 0000007372c20be0  x17 0000007371171358  x18 00000000ebad6082  x19 00000072ed214c00
05-27 05:16:11.931 26712 26712 F DEBUG   :     x20 0000007372b03590  x21 0000007373a6c5e0  x22 0000000000000000  x23 0000000012c242b8
05-27 05:16:11.931 26712 26712 F DEBUG   :     x24 0000000012c24718  x25 0000000012c24340  x26 00000072ed214ca0  x27 0000000000000000
05-27 05:16:11.931 26712 26712 F DEBUG   :     x28 0000000000000000  x29 0000007ff6196f20
05-27 05:16:11.931 26712 26712 F DEBUG   :     sp  0000007ff6196db0  lr  0000007372b036b0  pc  0000007371171360
05-27 05:16:12.034 26712 26712 F DEBUG   :
05-27 05:16:12.034 26712 26712 F DEBUG   : backtrace:
05-27 05:16:12.035 26712 26712 F DEBUG   :     #00 pc 0000000000070360  /system/lib64/libc.so (setgid+8)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #01 pc 000000000012f6ac  /system/lib64/libandroid_runtime.so (android::NativeVerifySystemIdmaps(_JNIEnv*, _jclass*)+284)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #02 pc 00000000003dcbb0  /system/framework/arm64/boot-framework.oat (offset 0x3d8000) (android.content.res.AssetManager.nativeVerifySystemIdmaps [DEDUPED]+144)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #03 pc 00000000007cb22c  /system/framework/arm64/boot-framework.oat (offset 0x3d8000) (android.content.res.AssetManager.createSystemAssetsInZygoteLocked+108)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #04 pc 00000000007cb790  /system/framework/arm64/boot-framework.oat (offset 0x3d8000) (android.content.res.AssetManager.getSystem+80)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #05 pc 00000000007d0a54  /system/framework/arm64/boot-framework.oat (offset 0x3d8000) (android.content.res.Resources.<init>+324)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #06 pc 00000000007d0dfc  /system/framework/arm64/boot-framework.oat (offset 0x3d8000) (android.content.res.Resources.getSystem+124)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #07 pc 000000000088e9c4  /system/framework/arm64/boot-framework.oat (offset 0x3d8000) (android.app.ActivityThread.systemMain+148)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #08 pc 000000000055d04c  /system/lib64/libart.so (art_quick_invoke_static_stub+604)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #09 pc 00000000000cf760  /system/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+232)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #10 pc 00000000002823b8  /system/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+344)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #11 pc 000000000027c374  /system/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+948)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #12 pc 000000000052db9c  /system/lib64/libart.so (MterpInvokeStatic+204)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #13 pc 000000000054f514  /system/lib64/libart.so (ExecuteMterpImpl+14612)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #14 pc 00000000000007f0  /data/data/com.termux/files/usr/libexec/termux-x11/oat/arm64/loader.vdex (com.termux.x11.Loader.main+24)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #15 pc 0000000000255e70  /system/lib64/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.1473445885+496)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #16 pc 000000000051cdd8  /system/lib64/libart.so (artQuickToInterpreterBridge+1032)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #17 pc 0000000000565efc  /system/lib64/libart.so (art_quick_to_interpreter_bridge+92)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #18 pc 000000000055d04c  /system/lib64/libart.so (art_quick_invoke_static_stub+604)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #19 pc 00000000000cf760  /system/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+232)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #20 pc 00000000004633d0  /system/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #21 pc 0000000000463028  /system/lib64/libart.so (art::InvokeWithVarArgs(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, std::__va_list)+416)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #22 pc 000000000036685c  /system/lib64/libart.so (art::JNI::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, std::__va_list)+644)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #23 pc 00000000000b3860  /system/lib64/libandroid_runtime.so (_JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+120)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #24 pc 00000000000b371c  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::callMain(android::String8 const&, _jclass*, android::Vector<android::String8> const&)+332)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #25 pc 00000000000025b0  /system/bin/app_process64 (android::AppRuntime::onStarted()+64)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #26 pc 00000000003dcbb0  /system/framework/arm64/boot-framework.oat (offset 0x3d8000) (android.content.res.AssetManager.nativeVerifySystemIdmaps [DEDUPED]+144)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #27 pc 000000000055d04c  /system/lib64/libart.so (art_quick_invoke_static_stub+604)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #28 pc 00000000000cf760  /system/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+232)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #29 pc 00000000002823b8  /system/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+344)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #30 pc 000000000027c374  /system/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+948)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #31 pc 000000000052db9c  /system/lib64/libart.so (MterpInvokeStatic+204)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #32 pc 000000000054f514  /system/lib64/libart.so (ExecuteMterpImpl+14612)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #33 pc 0000000000dda3ba  /system/framework/boot-framework.vdex (com.android.internal.os.RuntimeInit.main+48)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #34 pc 0000000000255e70  /system/lib64/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.1473445885+496)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #35 pc 000000000051cdd8  /system/lib64/libart.so (artQuickToInterpreterBridge+1032)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #36 pc 0000000000565efc  /system/lib64/libart.so (art_quick_to_interpreter_bridge+92)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #37 pc 000000000055d04c  /system/lib64/libart.so (art_quick_invoke_static_stub+604)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #38 pc 00000000000cf760  /system/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+232)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #39 pc 00000000004633d0  /system/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #40 pc 0000000000463028  /system/lib64/libart.so (art::InvokeWithVarArgs(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, std::__va_list)+416)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #41 pc 000000000036685c  /system/lib64/libart.so (art::JNI::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, std::__va_list)+644)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #42 pc 00000000000b3860  /system/lib64/libandroid_runtime.so (_JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+120)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #43 pc 00000000000b625c  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::start(char const*, android::Vector<android::String8> const&, bool)+884)
05-27 05:16:12.035 26712 26712 F DEBUG   :     #44 pc 00000000000021a0  /system/bin/app_process64 (main+1200)
05-27 05:16:12.036 26712 26712 F DEBUG   :     #45 pc 00000000000ae760  /system/lib64/libc.so (__libc_init+88)
05-27 05:16:13.600 26699 26699 E AndroidRuntime: FATAL EXCEPTION: main
05-27 05:16:13.600 26699 26699 E AndroidRuntime: PID: 26699
05-27 05:16:13.600 26699 26699 E AndroidRuntime: java.lang.UnsatisfiedLinkError: dlopen failed: library "/data/app/com.termux.x11-tru9m7s7szKY8Ou2FH0AIA==/base.apk!/lib/arm64-v8a/libXlorie.so" not found
05-27 05:16:13.600 26699 26699 E AndroidRuntime:        at java.lang.Runtime.loadLibrary0(Runtime.java:1016)
05-27 05:16:13.600 26699 26699 E AndroidRuntime:        at java.lang.System.loadLibrary(System.java:1669)
05-27 05:16:13.600 26699 26699 E AndroidRuntime:        at com.termux.x11.CmdEntryPoint.<clinit>(CmdEntryPoint.java:95)
05-27 05:16:13.600 26699 26699 E AndroidRuntime:        at java.lang.Class.classForName(Native Method)
05-27 05:16:13.600 26699 26699 E AndroidRuntime:        at java.lang.Class.forName(Class.java:453)
05-27 05:16:13.600 26699 26699 E AndroidRuntime:        at com.termux.x11.Loader.main(Loader.java:59)
05-27 05:16:13.600 26699 26699 E AndroidRuntime:        at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
05-27 05:16:13.600 26699 26699 E AndroidRuntime:        at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:340)
05-27 05:16:13.603 26699 26699 E AndroidRuntime: Error reporting crash
05-27 05:16:13.603 26699 26699 E AndroidRuntime: java.lang.RuntimeException: Bad file descriptor
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at android.os.BinderProxy.transactNative(Native Method)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at android.os.BinderProxy.transact(Binder.java:1137)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at android.os.ServiceManagerProxy.getService(ServiceManagerNative.java:125)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at android.os.ServiceManager.rawGetService(ServiceManager.java:253)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at android.os.ServiceManager.getService(ServiceManager.java:124)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at android.app.ActivityManager$1.create(ActivityManager.java:4133)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at android.app.ActivityManager$1.create(ActivityManager.java:4130)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at android.util.Singleton.get(Singleton.java:34)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at android.app.ActivityManager.getService(ActivityManager.java:4126)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$KillApplicationHandler.uncaughtException(RuntimeInit.java:143)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1068)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1063)
05-27 05:16:13.603 26699 26699 E AndroidRuntime:        at java.lang.Thread.dispatchUncaughtException(Thread.java:1955)
--------- beginning of main
05-27 05:18:11.067  1111 26879 I OMXClient: IOmx service obtained
05-27 05:18:16.224  1111  1128 I com.termux: Background concurrent copying GC freed 18035(6MB) AllocSpace objects, 2(104KB) LOS objects, 18% free, 26MB/32MB, paused 738us total 102.821ms
05-27 05:18:18.152  1111  1111 V ViewRootImpl[TermuxActivity]: [ANR Warning]Input routeing takes more than 6000ms since 2023-05-27 02:34:17.135, this = com.mediatek.view.impl.ViewDebugManagerImpl@480a605
05-27 05:18:18.153  1111  1111 V ViewRootImpl[TermuxActivity]: Input event delivered to android.view.ViewRootImpl$EarlyPostImeInputStage@c5e03da at 2023-05-27 05:18:18.152
05-27 05:18:18.153  1111  1111 V ViewRootImpl[TermuxActivity]: Input event delivered to android.view.ViewRootImpl$ViewPostImeInputStage@8b0cee8 at 2023-05-27 05:18:18.152
05-27 05:18:18.153  1111  1111 V ViewRootImpl[TermuxActivity]: Input event delivered to android.view.ViewRootImpl$NativePostImeInputStage@dd0400b at 2023-05-27 05:18:18.152
05-27 05:18:18.153  1111  1111 V ViewRootImpl[TermuxActivity]: Input event delivered to android.view.ViewRootImpl$SyntheticInputStage@fa883a6 at 2023-05-27 05:18:18.152
05-27 05:18:18.153  1111  1111 V ViewRootImpl[TermuxActivity]: [ANR Warning]Input routeing takes more than 6000ms since 2023-05-27 02:34:17.135, this = com.mediatek.view.impl.ViewDebugManagerImpl@480a605
05-27 05:18:18.153  1111  1111 V ViewRootImpl[TermuxActivity]: Input event delivered to android.view.ViewRootImpl$EarlyPostImeInputStage@c5e03da at 2023-05-27 05:18:18.153
05-27 05:18:18.153  1111  1111 V ViewRootImpl[TermuxActivity]: Input event delivered to android.view.ViewRootImpl$ViewPostImeInputStage@8b0cee8 at 2023-05-27 05:18:18.153
05-27 05:18:18.153  1111  1111 V ViewRootImpl[TermuxActivity]: Input event delivered to android.view.ViewRootImpl$NativePostImeInputStage@dd0400b at 2023-05-27 05:18:18.153
05-27 05:18:18.153  1111  1111 V ViewRootImpl[TermuxActivity]: Input event delivered to android.view.ViewRootImpl$SyntheticInputStage@fa883a6 at 2023-05-27 05:18:18.153
05-27 05:19:54.966  1111  1111 D ViewRootImpl[PopupWindow:a983951]: hardware acceleration = true , fakeHwAccelerated = false, sRendererDisabled = false, forceHwAccelerated = false, sSystemRendererDisabled = false
05-27 05:19:54.978  1111  1111 D ViewRootImpl[PopupWindow:b492a24]: hardware acceleration = true , fakeHwAccelerated = false, sRendererDisabled = false, forceHwAccelerated = false, sSystemRendererDisabled = false
05-27 05:19:55.016  1111  1143 D Surface : Surface::connect(this=0x7ad28cc000,api=1)
05-27 05:19:55.016  1111  1111 D Surface : Surface::allocateBuffers(this=0x7ad28cc000)
05-27 05:19:55.020  1111  1143 D mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
05-27 05:19:55.035  1111  1111 D Surface : Surface::allocateBuffers(this=0x7ac92c4000)
05-27 05:19:55.035  1111  1143 D Surface : Surface::connect(this=0x7ac92c4000,api=1)
05-27 05:19:55.038  1111  1143 D mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
05-27 05:19:55.129  1111  1111 D ViewRootImpl[PopupWindow:e78ab36]: hardware acceleration = true , fakeHwAccelerated = false, sRendererDisabled = false, forceHwAccelerated = false, sSystemRendererDisabled = false
05-27 05:19:55.193  1111  1111 D Surface : Surface::allocateBuffers(this=0x7ac9a0e000)
05-27 05:19:55.193  1111  1143 D Surface : Surface::connect(this=0x7ac9a0e000,api=1)
05-27 05:19:55.195  1111  1143 D mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
05-27 05:19:55.788  1111  1143 D Surface : Surface::disconnect(this=0x7ac9a0e000,api=1)
05-27 05:19:55.789  1111  1143 D Surface : Surface::disconnect(this=0x7ac9a0e000,api=1)
05-27 05:19:55.789  1111  1143 W libEGL  : EGLNativeWindowType 0x7ac9a0e010 disconnect failed
05-27 05:19:55.789  1111  1111 D View    : [Warning] assignParent to null: this = android.widget.PopupWindow$PopupDecorView{9c44c2 V.E...... R......D 0,0-316,93}
05-27 05:19:56.982  1111  1111 D ViewRootImpl[PopupWindow:e78ab36]: hardware acceleration = true , fakeHwAccelerated = false, sRendererDisabled = false, forceHwAccelerated = false, sSystemRendererDisabled = false
05-27 05:19:57.011  1111  1143 D Surface : Surface::connect(this=0x7ad55a6000,api=1)
05-27 05:19:57.011  1111  1111 D Surface : Surface::allocateBuffers(this=0x7ad55a6000)
05-27 05:19:57.016  1111  1143 D mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
05-27 05:19:57.328  1111  1143 D Surface : Surface::disconnect(this=0x7ad55a6000,api=1)
05-27 05:19:57.331  1111  1143 D Surface : Surface::disconnect(this=0x7ad55a6000,api=1)
05-27 05:19:57.331  1111  1143 W libEGL  : EGLNativeWindowType 0x7ad55a6010 disconnect failed
05-27 05:19:57.332  1111  1111 D View    : [Warning] assignParent to null: this = android.widget.PopupWindow$PopupDecorView{ce69b6c V.E...... R......D 0,0-316,93}
05-27 05:19:59.711  1111  1143 D Surface : Surface::disconnect(this=0x7ad28cc000,api=1)
05-27 05:19:59.713  1111  1143 D Surface : Surface::disconnect(this=0x7ad28cc000,api=1)
05-27 05:19:59.713  1111  1143 W libEGL  : EGLNativeWindowType 0x7ad28cc010 disconnect failed
05-27 05:19:59.714  1111  1111 D View    : [Warning] assignParent to null: this = android.widget.PopupWindow$PopupDecorView{3134535 V.E...... R.....I. 0,0-69,35}
05-27 05:20:08.991  1111  1111 D ViewRootImpl[PopupWindow:e78ab36]: hardware acceleration = true , fakeHwAccelerated = false, sRendererDisabled = false, forceHwAccelerated = false, sSystemRendererDisabled = false
05-27 05:20:09.058  1111  1143 D Surface : Surface::connect(this=0x7aca633000,api=1)
05-27 05:20:09.065  1111  1111 D Surface : Surface::allocateBuffers(this=0x7aca633000)
05-27 05:20:09.071  1111  1143 D mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
05-27 05:20:12.057  1111  1143 D Surface : Surface::disconnect(this=0x7aca633000,api=1)
05-27 05:20:12.059  1111  1143 D Surface : Surface::disconnect(this=0x7aca633000,api=1)
05-27 05:20:12.059  1111  1143 W libEGL  : EGLNativeWindowType 0x7aca633010 disconnect failed
05-27 05:20:12.060  1111  1111 D View    : [Warning] assignParent to null: this = android.widget.PopupWindow$PopupDecorView{5580258 V.E...... R......D 0,0-316,93}
05-27 05:20:13.493  1111  1111 D ViewRootImpl[PopupWindow:e78ab36]: hardware acceleration =
```

**Comment by twaik** [comment]
You do not need adb for this. Logcat works in termux commandline and shows logs only for termux processes so it is even better..

**Comment by Anonymous2716** [comment]
Wait a minute I think I broke the apk when I unpacked and repacked the apk.

**Comment by Anonymous2716** [comment]
> Logcat works in termux commandline and shows logs only for termux processes so it is even better..

You are right.

But its all there. I may have to start over and re do everything I did.

```
    /data/app/com.termux.x11-tru9m7s7szKY8Ou2FH0AIA==/lib/arm64 ·
❯ l
total 27M
drwxr-xr-x 2 system system  4.0K May 27 00:02 .
drwxr-xr-x 3 system system  4.0K May 27 00:02 ..
-rwxr-xr-x 1 system system  2.3M May 26 18:00 libGL.so
-rwxr-xr-x 1 system system  9.8M May 26 18:00 libXlorie.so
-rwxr-xr-x 1 system system 1003K May 26 18:00 libc++_shared.so
-rwxr-xr-x 1 system system  4.8K May 26 18:00 libexec-helper.so
-rwxr-xr-x 1 system system  425K May 26 18:00 libglapi.so
-rwxr-xr-x 1 system system  9.8K May 26 18:00 libtermux.so
-rwxr-xr-x 1 system system  1.8M May 26 18:00 libxkbcomp.so
-rwxr-xr-x 1 system system   12M May 26 18:00 swrast_dri.so

```

**Comment by twaik** [comment]
You can try to rebuild termux-x11 with `./gradlew assembleDebug`. That's all I can offer you.

**Comment by Anonymous2716** [comment]
@twaik you are right . but I may have found the issue and it needs more researching. What I was trying to do is make the app share userid with termux (without compiling from source). Its not needed but I was curious. So after I re built the app using apktool I got error on installation. Here is what I had : https://github.com/iBotPeaches/Apktool/issues/3081

And so it fixed the installation but this issue I am having right now. But i just resigned the both loader and x11 apk without modifying and it works great.

I also found that in The loader apk extractNativeLibs is set to false maybe that's causing the issue (maybe I'm not sure.) . 

But I guess I will just rebuild from source with the modifications I need to make. Thanks for your time.
